internal/test_cases/single_line_output_test_case.go

```diff
@@ -54,9 +54,13 @@ func (t SingleLineOutputTestCase) Run(shell *shell_executable.ShellExecutable, l
 		if errors.Is(err, shell_executable.ErrConditionNotMet) {
 			return fmt.Errorf("Expected first line of output to end with '\\n' (newline), got %q", string(cleanedOutput))
 		} else if errors.Is(err, shell_executable.ErrProgramExited) {
-			return fmt.Errorf("Expected shell to be a long-running process, but it exited")
+			exitCode := shell.ExitCode()
+			if exitCode == -1 {
+				return fmt.Errorf("Expected first line of output to end with '\\n' (newline), got %q. Program is still running", string(cleanedOutput))
```

Comment:
```suggestion
				return fmt.Errorf("Expected first line of output to end with '\\n' (newline), got %q", string(cleanedOutput))
```

(This I think would be implied, can remove so the log isn't huge)